lnd_test.go

```diff
@@ -616,41 +616,75 @@ func testMultiHopPayments(net *networkHarness, t *harnessTest) {
 	}
 
 	assertAsymmetricBalance := func(node *lightningNode,
-		chanPoint *wire.OutPoint, localBalance,
+		chanPoint wire.OutPoint, localBalance,
 		remoteBalance int64) {
 
-		listReq := &lnrpc.ListChannelsRequest{}
-		resp, err := node.ListChannels(ctxb, listReq)
-		if err != nil {
-			t.Fatalf("unable to for node's channels: %v", err)
+		channelName := ""
+		switch chanPoint {
+		case carolFundPoint:
+			channelName = "Carol(local) => Alice(remote)"
+		case aliceFundPoint:
+			channelName = "Alice(local) => Bob(remote)"
 		}
-		for _, channel := range resp.Channels {
-			if channel.ChannelPoint != chanPoint.String() {
-				continue
+
+		checkBalance := func() error {
+			listReq := &lnrpc.ListChannelsRequest{}
+			resp, err := node.ListChannels(ctxb, listReq)
+			if err != nil {
+				return fmt.Errorf("unable to for node's "+
+					"channels: %v", err)
+			}
+			for _, channel := range resp.Channels {
+				if channel.ChannelPoint != chanPoint.String() {
+					continue
+				}
+
+				if channel.LocalBalance != localBalance {
+					return fmt.Errorf("%v: incorrect local "+
+						"balances: %v != %v", channelName,
+						channel.LocalBalance, localBalance)
+				}
+
+				if channel.RemoteBalance != remoteBalance {
+					return fmt.Errorf("%v: incorrect remote "+
+						"balances: %v != %v", channelName,
+						channel.RemoteBalance, remoteBalance)
+				}
+
+				return nil
 			}
+			return fmt.Errorf("channel not found")
+		}
+
+		// As far as HTLC inclusion in commitment transaction might be
+		// postponed we will try to check the balance couple of
+		// times, and then if after some period of time we receive wrong
+		// balance return the error.
+		// TODO(roasbeef): remove sleep after invoice notification hooks
+		// are in place
+		var timeover bool
```

Comment:
This seems a bit fragile. There's also a race-condition here between this goroutine and the one checking the timeout value below. 

Switching to something like this should eliminate the race-condition: 
```go
var timeover uint32
go func() {
    <-time.After(time.Second * 10)
    atomic. StoreUint32(&timeover, 1)
}()
```
Then use `atomic. LoadUint32(&timeover)` in the main retry loop.